scripts/scan-pdf-content.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/env python3
+"""
+Scan user uploaded PDFs in S3 and flag ones which contain any
+non-text or image content (ie. JS or videos).
+
+Requires a running version of veraPDF-REST (https://github.com/veraPDF/veraPDF-rest)
+
+Usage: scan-pdf-content.py <stage> <framework> <verapdf-url> [--index=<index>]
+
+Options:
+    <stage>                         Stage to target
+    <framework>                     Slug for the framework to scan
+    <verapdf-url>                   The url for the verapdf-rest service to use
+    --index=<index>                 Optional. If supplied, the scan resumes from a previous run, using
+                                        the list of PDFs stored in '/tmp/pdf-scan-queue.json' starting at `index`
+
+    -h, --help                      Show this information
+
+Examples:
+    ./scripts/scan-pdf-content.py staging g-cloud-12 http://localhost:8080
+"""
+import boto3
+import requests
+import csv
+import uuid
+import logging
+import json
+import os
+import threading
+import queue
+from docopt import docopt
+from datetime import datetime
+
+from typing import Tuple, NamedTuple, Optional
+
+
+class ScanResult(NamedTuple):
+    scanned_at: str
+    status_code: int
+    message: str
+    bucket: str
+    framework: str
+    key: str
+
+
+def get_aws_session(stage: str) -> boto3.session.Session:
+    """
+    Start an authenticated session for the supplied stage.
+
+    :param stage: The stage we need credentials for. Must be one of 'production',
+        'staging' or 'preview'.
+    :return: An authenticated boto3.session.Session object
+    """
+    # production and staging data is in the production AWS account
+    if stage == "preview":
+        arn = "arn:aws:iam::381494870249:role/infrastructure"
+    else:
+        arn = "arn:aws:iam::050019655025:role/infrastructure"
```

Comment:
Are these the best roles to use for a developer to access S3? I got a permission error when I tried to use the `developer` role, but that might be caused by the limited permissions on my account. 

I looked in the `dm-aws` repo but couldn't see where the policy for the `developer` role was defined to check.

Reply:
I think normally we don't hard code these, instead we set `AWS_PROFILE` env var to direct which account to use.

Reply:
Ah I think I understand. That'll clean up the code quite a bit if I don't need to pass the authenticated `session` object around. Thanks!